### PR TITLE
GitHub Actions integration and packaging improvements

### PR DESCRIPTION
**Pull request by cryptaliagy**
Gonna list the basic changes:

* Adds GitHub Actions testing for all supported versions of python across all supported platforms
* Moved the uoapi folder to a src folder
* Updated MANIFEST.in file to just graft the src directory
* Updated setup.py file to auto-detect packages and other general cleanup
* Added test extras
* Added setup.cfg file to configure some tools such as linting and testing
* Added status badges to the README file

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,12 +1,138 @@
...
+dmypy.json
+
+# Pyre type checker
+.pyre/
+
+# pytype static type analyzer
+.pytype/
+
+# Cython debug symbols
+cython_debug/
```
Comment (andrewnags) [inline .gitignore:138]:
```suggestion
cython_debug/

# Vim
*.swp

# ctags/etags
tags
TAGS
```

**Review by andrewnags — CHANGES_REQUESTED** [review]
This looks great! Thank you so much!  I just wanted to keep the .gitignore entries for my development stuff, but other than that it's approved :)

File: .gitignore (hunk 1 above)
Comment (cryptaliagy) [reply .gitignore:138]: I'll add those but editor-specific files should generally exist inside a global gitignore: https://gist.github.com/subfuzion/db7f57fff2fb6998a16c

Comment (andrewnags) [reply .gitignore:138]: Neat, I didn't know you could do that!  Thanks anyways!

**Review by andrewnags — APPROVED** [review]

**Comment by andrewnags** [comment]
Not sure what happened, but looks like the mac and windows builds are failing? For windows seems like pip needs a `--user` to avoid needing admin permissions, but not sure about mac.

**Pull request merged**